pkgs/development/tools/ocaml/opam/default.nix

```diff
@@ -47,7 +47,8 @@ in stdenv.mkDerivation rec {
   name = "opam-${version}";
   version = "1.2.2";
 
-  buildInputs = [ unzip curl ncurses ocaml makeWrapper];
+  buildInputs = [ ncurses ocaml makeWrapper ];
+  propagatedBuildInputs = [ unzip curl ];
```

Comment:
`propagatedBuildInputs` are only meant for build dependencies and they work by accident in `nix-shell` too.  Use `--suffix PATH : ${curl}/bin` for `makeWrapper` below instead.